frontend/public/components/monitoring/alert-manager-receiver-forms.tsx

```diff
@@ -200,143 +210,136 @@ const RoutingLabels = ({ routeLabels, setRouteLabels, saveAsDefaultReceiver }) =
   );
 };
 
-const ReceiverBaseForm: React.FC<ReceiverBaseFormProps> = ({ obj, titleVerb, saveButtonText }) => {
+const ReceiverBaseForm: React.FC<ReceiverBaseFormProps> = ({
+  obj,
+  titleVerb,
+  saveButtonText,
+  editReceiverNamed,
+}) => {
   const secret: K8sResourceKind = obj; // Secret "alertmanager-main" which contains alertmanager.yaml config
   const [errorMsg, setErrorMsg] = React.useState('');
   const [inProgress, setInProgress] = React.useState(false);
   let config: AlertManagerConfig;
   if (!errorMsg) {
     config = getAlertManagerConfig(secret, setErrorMsg);
   }
-  const [receiverName, setReceiverName] = React.useState('');
-  const [receiverType, setReceiverType] = React.useState('');
-  const [webhookUrl, setWebhookUrl] = React.useState('');
-  const defaultPagerDutyValues = {
-    integrationType: 'events',
+
+  let initialReceiverName = '';
+  let initialReceiverType = '';
+  let initialPagerDutyValues = {
+    integrationType: 'events', // 'Events API v2'
     integrationKey: '',
   };
-  const [pagerDutyValues, setPagerDutyValues] = React.useState(defaultPagerDutyValues);
+  let initialWebhookUrl = '';
+
+  let receiverToEdit: AlertManagerReceiver;
+  if (editReceiverNamed) {
+    receiverToEdit = _.find(_.get(config, 'receivers'), { name: editReceiverNamed });
+    initialReceiverName = receiverToEdit ? receiverToEdit.name : '';
+    initialReceiverType = _.find(_.keys(receiverToEdit), (key) => _.includes(key, '_configs'));
+    if (initialReceiverType) {
+      let receiverConfig;
+      initialReceiverType = initialReceiverType.slice(0, initialReceiverType.indexOf('_'));
```

Comment:
I think it's probably better to not bother with the `slice` and instead just do `case 'pagerduty_configs'`, etc.

Then we can be sure we're matching just `pagerduty_configs`, not `/pagerduty_.*_configs/`.

Reply:
This resulted in a few more changes in other areas, but I think it's a little better.  Pls see last commit.